src/libLookup/Lookup.cpp

```diff
@@ -1901,6 +2085,69 @@ bool Lookup::ProcessSetStateDeltaFromSeed(const bytes& message,
   return true;
 }
 
+bool Lookup::ProcessSetStateDeltasFromSeed(const bytes& message,
+                                           unsigned int offset,
+                                           const Peer& from) {
+  LOG_MARKER();
+
+  if (AlreadyJoinedNetwork()) {
+    cv_setStateDeltasFromSeed.notify_all();
+    return true;
+  }
+
+  unique_lock<mutex> lock(m_mutexSetStateDeltasFromSeed);
+
+  uint64_t lowBlockNum = 0;
+  uint64_t highBlockNum = 0;
+  vector<bytes> stateDeltas;
+  PubKey lookupPubKey;
+
+  if (!Messenger::GetLookupSetStateDeltasFromSeed(message, offset, lowBlockNum,
+                                                  highBlockNum, lookupPubKey,
+                                                  stateDeltas)) {
+    LOG_EPOCH(WARNING, m_mediator.m_currentEpochNum,
+              "Messenger::GetLookupSetStateDeltasFromSeed failed.");
+    return false;
+  }
+
+  if (!VerifySenderNode(GetSeedNodes(), lookupPubKey)) {
+    LOG_EPOCH(WARNING, m_mediator.m_currentEpochNum,
+              "The message sender pubkey: "
+                  << lookupPubKey << " is not in my lookup node list.");
+    return false;
+  }
+
+  LOG_EPOCH(INFO, m_mediator.m_currentEpochNum,
+            "ProcessSetStateDeltasFromSeed sent by "
+                << from << " for blocks: " << lowBlockNum << " to "
+                << highBlockNum);
+
+  for (const auto& delta : stateDeltas) {
```

Comment:
I think we need to verify the state delta against the state delta hash in the corresponding txblock. One way to locate the txBlock is to incrementally using the count in the vector and the lowBlockNum. Another way can be modifying the message structure to indicate the block number of each delta, which may require extra code change however but it's better for future maintenance.

Reply:
Since, we check on the count of deltas received against the request low/hignum and also verify the sender (lookup), additional check on state-delta hash can be considered for future improvements. If you agree.

Reply:
Yes, we can do that enhancement in the future